### PR TITLE
Change from Jekyll to 11ty

### PR DESCRIPTION
**Pull request by bjohansebas**
Currently, Jekyll is a good static site generator, but there are now better tools like 11ty that can greatly improve the development experience. This includes enhancements in compilation time, the implementation of tools like vitejs, and even the minification of pages.

Several of the changes do not require rewriting the documentation, as 11ty supports Liquid, facilitating the transition to newer technologies that enhance the documentation experience such as mdx.

<details><summary>Build time</summary>
<p>

</p>
</details> 

TODO: 

- [ ] Update write posts
- [ ] Update readme

### CODE REVIEW

File: en/blog/write-post.md [hunk 1]
```diff
@@ -6,24 +6,23 @@ lang: en
...
+![Blogger](/images/blogger.jpg)
 
 If you have an idea for a blog post, follow these steps to propose it and potentially get it published!
 
 1. ### Propose your post
 Before taking the time to write a post, please confirm that we will be able to publish it. We're looking for topics specifically related to Express, and so we want to pre-approve all posts. For the moment, this means we aren't accepting any unsolicited posts. To propose a blog post, [open an issue](https://github.com/expressjs/expressjs.com/issues) entitled  `Blog post proposal: <your idea>`.
 
 
-1. ### Fork the repository 
+2. ### Fork the repository
```
Comment (ctcpip) [inline en/blog/write-post.md:17]: these should all remain `1.`

**Comment by crandmck** [comment]
I think it's reasonable to move from Jekyll to a more modern framework, however I think a change of this magnitude requires at least _some_ discussion.  While 11ty did come up in conversation, I don't think we've had any specific discussion and evaluation of other frameworks in an issue... correct me if I'm wrong.

Before landing this, I'd like to at least discuss the framework to which we'll move and ideally get consensus on a decision.
We can use any of these issues:
- https://github.com/expressjs/expressjs.com/issues/1206
- https://github.com/expressjs/discussions/issues/211

**Comment by UlisesGascon** [comment]
I agree with @crandmck! We might want to choose the right tooling first. Something that might be a good fit for us is https://docusaurus.io/. It already has many features we need out of the box, like MDX, i18n, blog, responsive design, dark theme, API Docs versioning, etc. It's widely used by JS libraries like Jest (https://jestjs.io/).

I think one of the keys for us is to try to use a framework that allows us to focus on minor changes and content rather than building a lot of components, pages, etc., from scratch.

**Comment by bjohansebas** [comment]
@UlisesGascon @crandmck Of course, you are right; it is hasty to switch from one tool to another, and it requires discussions to ensure that nothing in the documentation breaks

Although I had also chosen this library to facilitate an easy transition that supports Liquid and doesn’t require much modification to the documentation, with the intention of later migrating to other tools like Starlight, VitePress, or Docusaurus, after having converted the documentation to MDX, which is used by these tools as a content generator.

**Pull request closed without merging**